src/pytorch_lightning/utilities/device_parser.py

```diff
@@ -330,3 +332,13 @@ def parse_hpus(devices: Optional[Union[int, str, List[int]]]) -> Optional[int]:
         raise MisconfigurationException("`devices` for `HPUAccelerator` must be int, string or None.")
 
     return int(devices) if isinstance(devices, str) else devices
+
+
+def num_cuda_devices() -> int:
+    with multiprocessing.Pool(1) as pool:
+        return pool.apply(torch.cuda.device_count)
+
+
+def is_cuda_available() -> bool:
+    with multiprocessing.Pool(1) as pool:
+        return pool.apply(torch.cuda.is_available)
```

Comment:
```suggestion
# the following functions are run in a separate process to avoid creating 
# a cuda context in the main process where we later might want to fork. 
def num_cuda_devices() -> int:
    with multiprocessing.Pool(1) as pool:
        return pool.apply(torch.cuda.device_count)


def is_cuda_available() -> bool:
    with multiprocessing.Pool(1) as pool:
        return pool.apply(torch.cuda.is_available)
```